alcs-frontend/src/app/shared/header/header.component.ts

```diff
@@ -12,27 +16,109 @@ import { AuthenticationService } from '../../services/authentication/authenticat
 })
 export class HeaderComponent implements OnInit {
   homeUrl = environment.homeUrl;
-  decisionMakers: ApplicationDecisionMakerDto[] = [];
+  currentUser!: ICurrentUser;
+  currentUserProfile?: UserDto;
+  private decisionMakers: ApplicationDecisionMakerDto[] = [];
+  $sortedDecisionMakers!: Observable<ApplicationDecisionMakerDto[]>;
 
   constructor(
     private authService: AuthenticationService,
     private applicationService: ApplicationService,
+    private toastService: ToastService,
+    private userService: UserService,
     private router: Router
   ) {}
 
   ngOnInit(): void {
-    this.applicationService.$applicationDecisionMakers.subscribe((dms) => {
-      this.decisionMakers = dms;
+    this.authService.isAuthenticated.subscribe((isAuthenticated) => {
+      if (isAuthenticated) {
+        this.currentUser = this.authService.getCurrentUser();
+
+        this.userService.fetchUsers();
+        this.applicationService.setup();
+      }
+    });
+
+    this.userService.$users.subscribe((users) => {
+      this.currentUserProfile = users.find((u) => u.email === this.currentUser.email);
     });
+
+    this.$sortedDecisionMakers = this.applicationService.$applicationDecisionMakers.pipe(
+      map((dms) => this.transformDecisionMakers(dms))
+    );
+  }
+
+  private transformDecisionMakers(dms: ApplicationDecisionMakerDto[]) {
+    this.decisionMakers = dms
+      .map((dm) => {
+        return {
+          ...dm,
+          isFavorite: this.currentUserProfile?.settings?.favoriteBoards?.includes(dm.code),
+        };
+      })
+      .sort((x, y) => this.sortDecisionMakers(x, y));
+    return this.decisionMakers;
+  }
+
+  private sortDecisionMakers(x: ApplicationDecisionMakerDto, y: ApplicationDecisionMakerDto) {
+    if (x.isFavorite && !y.isFavorite) {
+      return -1;
+    }
+
+    if (y.isFavorite && !x.isFavorite) {
+      return 1;
+    }
+
+    if (x.isFavorite === y.isFavorite) {
+      if (x.label > y.label) return 1;
+      if (x.label < y.label) return -1;
+    }
+    return 0;
   }
 
-  onSelectBoard(event: MouseEvent) {
-    const target = event.currentTarget as HTMLInputElement;
-    const boardCode = target.value;
-    this.router.navigateByUrl(`/board/${boardCode}`);
+  onSelectBoard(dmCode: string) {
+    this.router.navigateByUrl(`/board/${dmCode}`);
   }
 
   onLogout() {
     this.authService.logout();
   }
+
+  private updateFavoriteBoardsList(dm: ApplicationDecisionMakerDto) {
+    if (!this.currentUserProfile) {
```

Comment:
The parent function has already run this check and its only called from that one function, can this check be removed?

Reply:
this is a way to avoid TS warnings about possible null. Another way would be to force it everywhere with '!' but it looks more correct to me to have this check.

Reply:
That makes sense